build_tools/github_actions/configure_ci.py

```diff
@@ -8,43 +8,76 @@
 """Determines whether CI should run on a given PR.
 
 The following environment variables are required:
-- GITHUB_EVENT_NAME: GitHub event name, e.g. pull_request.
-- GITHUB_OUTPUT: path to write workflow output variables.
-- GITHUB_STEP_SUMMARY: path to write workflow summary output.
+"GITHUB_EVENT_NAME",: GitHub event name, e.g. pull_request.
+"GITHUB_OUTPUT",: path to write workflow output variables.
+"GITHUB_STEP_SUMMARY",: path to write workflow summary output.
 
 When GITHUB_EVENT_NAME is "pull_request", there are additional environment
 variables to be set:
-- PR_BRANCH (required): PR source branch.
-- PR_TITLE (required): PR title.
-- PR_BODY (optional): PR description.
-- PR_LABELS (optional): JSON list of PR label names.
-- BASE_REF (required): base commit SHA of the PR.
-- ORIGINAL_PR_TITLE (optional): PR title from the original PR event, showing a
+"PR_BRANCH", (required): PR source branch.
+"PR_TITLE", (required): PR title.
+"PR_BODY", (optional): PR description.
+"PR_LABELS", (optional): JSON list of PR label names.
+"BASE_REF", (required): base commit SHA of the PR.
+"ORIGINAL_PR_TITLE", (optional): PR title from the original PR event, showing a
     notice if PR_TITLE is different.
-- ORIGINAL_PR_BODY (optional): PR description from the original PR event,
+"ORIGINAL_PR_BODY", (optional): PR description from the original PR event,
     showing a notice if PR_BODY is different. ORIGINAL_PR_TITLE must also be
     set.
-- ORIGINAL_PR_LABELS (optional): PR labels from the original PR event, showing a
+"ORIGINAL_PR_LABELS", (optional): PR labels from the original PR event, showing a
     notice if PR_LABELS is different. ORIGINAL_PR_TITLE must also be set.
 
 Exit code 0 indicates that it should and exit code 2 indicates that it should
 not.
 """
 
 import difflib
+import enum
 import fnmatch
 import json
 import os
 import re
+import string
 import subprocess
+import sys
 import textwrap
-from typing import Iterable, List, Mapping, Sequence, Tuple
+from typing import Iterable, List, Mapping, Sequence, Set, Tuple
+
+import yaml
+
+
+# We don't get StrEnum till Python 3.11
+@enum.unique
+class Trailer(str, enum.Enum):
+    __str__ = str.__str__
+
+    SKIP_CI = "skip-ci"
+    SKIP_JOBS = "ci-skip"
+    EXTRA_JOBS = "ci-extra"
+    EXACTLY_JOBS = "ci-exactly"
+    RUNNER_ENV = "runner-env"
+    BENCHMARK_EXTRA = "benchmark-extra"
+    # Trailer to prevent benchmarks from always running on LLVM integration PRs.
+    SKIP_LLVM_INTEGRATE_BENCHMARK = "skip-llvm-integrate-benchmark"
```

Comment:
Let's document these somewhere easy to point to... maybe https://github.com/openxla/iree/tree/main/.github/workflows/README.md ?

Reply:
How about in a developer guide linked from the CONTRIBUTING.md?

Reply:
I'm a bit hesitant to add text to CONTRIBUTING.md (I remember it triggering a banner for all developers saying something changed). I also think this optionality is something that _most_ developers should never need to use.

However, adding a link to a developer guide and then putting the "power PR author" features in that guide seems reasonable.